### PR TITLE
deviantArt provider

### PR DESCRIPTION
**Pull request by tparnell8**

### CODE REVIEW

**Comment by tparnell8** [comment]
#4

File: src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationDefaults.cs [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+        /// <summary>
+        /// Default value for <see cref="OAuthAuthenticationOptions.CallbackPath"/>.
+        /// </summary>
+        public const string CallbackPath = "/signin-deviantart";
+
+        /// <summary>
+        /// Default value for <see cref="OAuthAuthenticationOptions.AuthorizationEndpoint"/>.
+        /// </summary>
+        public const string AuthorizationEndpoint = "https://www.DeviantArt.com/oauth2/authorize";
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationDefaults.cs]: Please fix the casing of these URLs :smile:

File: src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+namespace AspNet.Security.OAuth.DeviantArt {
+    public class DeviantArtAuthenticationHandler : OAuthAuthenticationHandler<DeviantArtAuthenticationOptions> {
+        public DeviantArtAuthenticationHandler([NotNull] HttpClient client)
+            : base(client) {
+        }
+
+        protected override async Task<AuthenticationTicket> CreateTicketAsync([NotNull] ClaimsIdentity identity,
+            [NotNull] AuthenticationProperties properties, [NotNull] OAuthTokenResponse tokens) {
+            var request = new HttpRequestMessage(HttpMethod.Get, $"{Options.UserInformationEndpoint}?access_token={tokens.AccessToken}");
```
Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: I guess this is needed as DeviantArt doesn't support the Authorization header? There's nothing about that in the documentation...

Comment (tparnell8) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: I havn't seen anything in their docs. I looked for it, unless you can find something.

Comment (PinpointTownes) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: And have you tried using the Authorization header?

Comment (tparnell8) [inline src/AspNet.Security.OAuth.Providers.DeviantArt/DeviantArtAuthenticationHandler.cs]: no, but I guess I can try later on...should it just be Authorization: <OAuthTokenHere>

**Comment by tparnell8** [comment]
@PinpointTownes the header did work, also I fixed the urls

**Pull request closed without merging**